app/services/order_syncer.rb

```diff
@@ -106,4 +110,22 @@ def force_ship_address_required?(order)
       order.ship_address[attr] == distributor_address[attr]
     end
   end
+
+  def select_shipping_method_for_shipment(shipment, shipping_method)
+    shipping_rate = find_or_create_shipping_rate_for_shipping_method(shipment, shipping_method)
+    shipment.update_attributes(selected_shipping_rate_id: shipping_rate.id)
+  end
+
+  def find_or_create_shipping_rate_for_shipping_method(shipment, shipping_method)
+    shipping_rate = shipment.shipping_rates.where(shipping_method_id: shipping_method.id).last
```

Comment:
as in order_shipment.select_shipping_method you can do:
shipping_rate = shipment.shipping_rates.find_by_shipping_method_id(shipping_method_id)